operator/pkg/apis/istio/v1alpha1/values_types.proto

```diff
@@ -636,11 +636,13 @@ message PilotConfig {
   // Examples: 300s, 30m, 1h
   google.protobuf.Duration keepaliveMaxServerConnectionAge = 13;
 
-  // Labels that are added to Pilot pods.
+  // Labels that are added to Pilot deployment and pods.
   //
   // See https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/
   TypeMapStringInterface deploymentLabels = 14;
 
+  TypeMapStringInterface selectorLabels = 36;
```

Comment:
IMO this is dangerous and not a good practice. Selector labels are immutable. 

We can add pod labels without making them selector labels.

Reply:
@howardjohn could you please share your thoughts, what'll be the way to add labels for pilot.  The improvement adds labels for pod which allows it on gitops tools like ArgoCD. In my case, I tried to make the changes for the pods after provisioning, but argoCD rollback.

Reply:
@howardjohn You are completely right - I'm team member of @vedmichv and we found out yesterday by issue on prod that [Istio gateway spec](https://istio.io/latest/docs/reference/config/istio.operator.v1alpha1/#GatewaySpec) `labels` are actually selector + pod labels which is not clearly documented. When we've tried to add a new label Istio operator was not able to replace / patch manifest and ended in error state.